pkg/sqlite/configmap_writer.go

```diff
@@ -0,0 +1,214 @@
+	clientset, err := client.NewKubeClient(kubeconfig, logrus.StandardLogger())
+	if err != nil {
+		logrus.Fatalf("cluster config failed: %v", err)
+	}
+
+	return &ConfigMapWriter{
+		manifestsDir:  manifestsDir,
+		configMapName: configMapName,
+		namespace:     namespace,
+		clientset:     clientset,
+	}
+}
+
+func TranslateInvalidChars(input string) string {
+	validConfigMapKey := unallowedKeyChars.ReplaceAllString(input, "~")
+	return validConfigMapKey
+}
+
+func (c *ConfigMapWriter) Populate() error {
+	subDirs := []string{"manifests/", "metadata/"}
+
+	configMapPopulate, err := c.clientset.CoreV1().ConfigMaps(c.namespace).Get(c.configMapName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	configMapPopulate.Data = map[string]string{}
+
+	var totalSize uint64
+	for _, dir := range subDirs {
+		completePath := c.manifestsDir + dir
+		files, err := ioutil.ReadDir(completePath)
+		if err != nil {
+			logrus.Errorf("read dir failed: %v", err)
+			return err
+		}
+
+		for _, file := range files {
+			log := logrus.WithField("file", completePath+file.Name())
+			log.Info("Reading file")
+			content, err := ioutil.ReadFile(completePath + file.Name())
+			if err != nil {
+				log.Errorf("read failed: %v", err)
+				return err
+			}
+			totalSize += uint64(len(content))
+			if totalSize > (1 << 20) {
+				// over 1 MiB limit, bail out
+				log.Errorf("File with size %v exceeded 1 MiB limit, aboring", len(content))
+				return fmt.Errorf("file %v bigger than total allowed limit", file.Name())
+			}
+
+			validConfigMapKey := TranslateInvalidChars(file.Name())
+			if validConfigMapKey != file.Name() {
+				logrus.Warningf("translated filename from '%v' to '%v' for configmap compatability", file.Name(), validConfigMapKey)
+			}
+			if file.Name() == AnnotationsFilename {
+				var annotationsFile AnnotationsFile
+				err := yaml.Unmarshal(content, &annotationsFile)
+				if err != nil {
+					return err
+				}
+				configMapPopulate.SetAnnotations(map[string]string{
+					ResourceKey:  annotationsFile.Annotations.Resources,
+					MediaTypeKey: annotationsFile.Annotations.MediaType,
+				})
+			} else {
+				configMapPopulate.Data[validConfigMapKey] = string(content)
+			}
+		}
+	}
+
+	if sourceImage := os.Getenv(EnvContainerImage); sourceImage != "" {
+		annotations := configMapPopulate.GetAnnotations()
+		annotations[ConfigMapImageAnnotationKey] = sourceImage
+	}
+
+	_, err = c.clientset.CoreV1().ConfigMaps(c.namespace).Update(configMapPopulate)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// LaunchBundleImage will launch a bundle image and also create a configmap for
+// storing the data that will be updated to contain the bundle image data. It is
+// the responsibility of the caller to delete both the job and configmap when done.
+func LaunchBundleImage(kubeclient kubernetes.Interface, bundleImage, initImage, namespace string) (*corev1.ConfigMap, *batchv1.Job, error) {
```

Comment:
The intention is that this is called from OLM, correct?

Reply:
Yes!

Reply:
Changed the comment to be more clear about the intention.